joemcgill/gutenberg
# Add initial XHProf support to wp-env

## What?


This updates the configuration scripts to support passing a new --xdebug flag to wp env start, which will install and configure XHProf support to the WordPress environment. To enable profiling, you'll need to edit the wp-config.php file to include the following code where the `wp-settings.php` file is included:

```php
// start profiling
xhprof_enable();

/** Sets up WordPress vars and included files. */

require_once ABSPATH . 'wp-settings.php';

// stop profiler
$xhprof_data = xhprof_disable();

// display raw xhprof data for the profiler run
print_r($xhprof_data);
```

For more info about how to configure XHProf, see: https://github.com/longxinH/xhprof.
Also see the original Facebook docs for XHProf here: http://web.archive.org/web/20110514095512/http://mirror.facebook.net/facebook/xhprof/doc.html.



## Why?


XHProf is an open source application profiler for PHP. This would allow anyone using the @wordpress/env package to start an environment with the profiler configured in order to more easily profile their application.

## How?


This installs XHProf to the docker environment from PECL during setup and includes the correct `php.ini` configuration so that `XHProf` is available to the applocation.

## Testing Instructions


To test this experiment, you'll need to clone this repo, and check out this branch. Then cd to the `packages/env` directory and use `npm link` to create a symlink for the `@wordpress/env` package from this fork. Then, in the project where you want to test this package, run `npm link @wordpress/env` to use the symlinked version of the package and run `npm run wp-env start -- --xhprof` to start the environment with XHProf enabled.

You'll then need to edit your environment's wp-config.php file to include the snippet above in the PR description.

### Testing Instructions for Keyboard


## Screenshots or screencast
